src/interaction/interaction_elph.h

```diff
@@ -27,50 +32,52 @@
  * 3) for polar materials, adds the long-range Frohlich interaction.
  */
 // TODO: add flag to let user decide whether to use or not polar corrections
-class InteractionElPhWan {
+class InteractionElPhWan: public InteractionElPhBase {
+
+private:
 
-  Crystal crystal;
-  PhononH0 phononH0;
-  int numPhBands, numElBands, numWsR1Vectors, numWsR2Vectors;
-  std::vector<Eigen::Tensor<double, 3>> cacheCoupling;
-  bool usePolarCorrection = false;
-  const int phaseConvention;
+    ComplexView4D elPhCached;
+    ComplexView5D couplingWannier_k;
 
-  // kokkos objects for GPU accelerated elph coupling
-  ComplexView4D elPhCached;
-  ComplexView5D couplingWannier_k;
-  DoubleView2D wsR2Vectors_k;
-  DoubleView1D wsR2VectorsDegeneracies_k;
-  DoubleView2D wsR1Vectors_k;
-  DoubleView1D wsR1VectorsDegeneracies_k;
-  std::vector<ComplexView4D::HostMirror> elPhCached_hs;
+public:
 
-  // TODO TEMPORARY, for old interpolation method
-  Eigen::MatrixXd wsR1Vectors;
-  Eigen::VectorXd wsR1VectorsDegeneracies;
-  Eigen::MatrixXd wsR2Vectors;
-  Eigen::VectorXd wsR2VectorsDegeneracies;
-  Eigen::Tensor<std::complex<double>, 4> elPhCached_old;
-  Eigen::Vector3d cachedK1;
-  Eigen::Tensor<std::complex<double>, 5> couplingWannier;
+    // InteractionElPhWan() = default; // Default constructor
+    InteractionElPhWan(Crystal &crystal_, PhononH0 *phononH0_ = nullptr)
+            : InteractionElPhBase(crystal_, phononH0_) {}
+
+    /** Add polar correction to the electron-phonon coupling.
+    * @param q3: phonon wavevector, in cartesian coordinates
+    * @param ev1: eigenvector (rotation matrix U) at k
+    * @param ev2: eigenvector (rotation matrix U) at k'
+    * @param ev3: phonon eigenvector at q = k'-k
+    * @return g^L: the long-range (Frohlich) component of the el-ph interaction,
+    * as a tensor of shape (nb1,nb2,numPhBands)
+    */
+/*     Eigen::Tensor<std::complex<double>, 3>
+    getPolarCorrection(const Eigen::Vector3d &q3, const Eigen::MatrixXcd &ev1,
+                        const Eigen::MatrixXcd &ev2, const Eigen::MatrixXcd &ev3);
+ */
+    /** Estimate the memory in bytes, occupied by the kokkos Views containing
+    * the coupling tensor to be interpolated.
+    *
+    * @return a memory estimate in bytes
+    */
+    double getDeviceMemoryUsage() const override; // Added const
 
+    // Override for getCouplingSquared from base class
+    const Eigen::Tensor<double, 3>& getCouplingSquared(const int &ik2) const override;
+
+    // Override for resetK1 from base class
+    void resetK1() override;
+
+    std::vector<ComplexView4D::HostMirror> elPhCached_hs;
 #ifdef MPI_AVAIL
-  std::vector<MPI_Request> mpi_requests;
+    std::vector<MPI_Request> mpi_requests;
 #endif
 
 public:
 
-  // phase convention options -- TODO switch to ENUM
-  // Giustino uses Re, Rp for R vectors
```

Comment:
This phase convention information is still needed and it's an important update to the code -- can you move these over into the base class?